### PR TITLE
refactor repository

### PR DESCRIPTION
**Pull request by WojciechKusa**
- update requirements
- improve installation instructions
- remove unused scripts
- test lexical and neural models

### CODE REVIEW

File: config/prediction_config.yml [hunk 1]
```diff
@@ -17,5 +17,5 @@ easy:
   ]
```
Comment (WojciechKusa) [inline config/prediction_config.yml]: @ospitia I think this config is not used? can we remove it?

File: config/train_config.yml [hunk 2]
```diff
@@ -24,8 +24,8 @@ easy:
       'detailed_description'
```
Comment (WojciechKusa) [inline config/train_config.yml]: @ospitia same here, I think this config is not used? can we remove it?

File: trec_cds/neural/utils/evaluator.py [hunk 3]
```diff
@@ -227,20 +216,8 @@ def write_run(self, run, n_queries=50):
         run["Q0"] = "Q0"
         run["run_id"] = self.run_id
         run["rank"] = list(range(1, 51)) * n_queries
```
Comment (WojciechKusa) [inline trec_cds/neural/utils/evaluator.py]: @ospitia this line might fail if `batch_size` is smaller than `n_queries`?

File: config/prediction_config.yml (hunk 1 above)
Comment (ospitia) [reply config/prediction_config.yml]: True, this was a dummy example

File: trec_cds/neural/utils/evaluator.py [hunk 4]
```diff
@@ -64,39 +53,38 @@ class Evaluator:
 
     def __init__(
         self,
-        optimization_metric=P@10,
+        optimization_metric=P @ 10,
```
Comment (ospitia) [inline trec_cds/neural/utils/evaluator.py:56]: is this still working? 😯

File: trec_cds/neural/utils/evaluator.py (hunk 3 above)
Comment (ospitia) [reply trec_cds/neural/utils/evaluator.py]: oh, this one is supposed to be used when you have the whole experiment

**Review by ospitia — APPROVED** [review]

File: trec_cds/neural/utils/evaluator.py (hunk 4 above)
Comment (WojciechKusa) [reply trec_cds/neural/utils/evaluator.py:56]: yes!

File: trec_cds/neural/utils/evaluator.py (hunk 3 above)
Comment (WojciechKusa) [reply trec_cds/neural/utils/evaluator.py]: ok, I will leave it like this for now but open another PR and try to add it as a parameter.

**Pull request merged**